todo_app/tests_e2e/test_selenium.py

```diff
@@ -0,0 +1,360 @@
+    thread = Thread(target=lambda: application.run(use_reloader=False))
+    thread.daemon = True
+    thread.start()
+    yield app
+    # Tear Down
+    thread.join(1)
+    delete_trello_board(board_id)
+
+def create_trello_board(board_name):
+    trello = Trello()
+    create_board_url = trello.add_key_and_token("https://api.trello.com/1/boards?name=" + board_name + "&")
+    request = requests.post(create_board_url)
+    response = request.json()
+    return response["id"]
+
+def get_list_id_for_board(board_id, list_name):
+    trello = Trello()
+    lists_url = trello.add_key_and_token("https://api.trello.com/1/boards/" + board_id + "/lists?")
+    request = requests.get(lists_url)
+    response = request.json()
+    for item in response:
+        if list_name.upper() == item["name"].upper():
+            list_id = item["id"]
+    return list_id
+
+def delete_trello_board(board_id):
+    trello = Trello()
+    delete_board_url = trello.add_key_and_token("https://api.trello.com/1/boards/" + board_id + "?")
+    requests.delete(delete_board_url)
+    return
+
+def test_task_journey(driver, test_app):
+    driver.get('http://localhost:5000/')
+
+    assert driver.title == 'To-Do App'
+
+def test_can_create_new_to_do_item(driver, test_app):
+    new_card_name = "New Selenium Test Item"
+    driver.get('http://localhost:5000/')
+    new_to_do_textbox = driver.find_element_by_name("todoTitle")
+    new_to_do_textbox.send_keys(new_card_name)
+    save_button = driver.find_element_by_id("save_new")
+    ActionChains(driver).click(save_button).perform()
+
+    element_present = expected_conditions.presence_of_element_located((By.XPATH, "//table[@id='to-do-table']/tbody/tr[1]"))
+    WebDriverWait(driver, 60).until(element_present)
+
+    to_do_list_table_rows = driver.find_element_by_xpath("//table[@id='to-do-table']/tbody").find_elements_by_tag_name("tr")
+    doing_list_table_rows = driver.find_element_by_xpath("//table[@id='doing-table']/tbody").find_elements_by_tag_name("tr")
+    done_list_table_rows = driver.find_element_by_xpath("//table[@id='done-table']/tbody").find_elements_by_tag_name("tr")
+    to_do_list_table_cell = driver.find_element_by_xpath("//table[@id='to-do-table']/tbody/tr[1]/td[1]")
+
+    #make sure there is one row in todo table and none in the doing and done
+    assert len(to_do_list_table_rows) == 1
+    assert len(doing_list_table_rows) == 0
+    assert len(done_list_table_rows) == 0
+    #make sure the todo table row has a value and the text is "New Selenium Test Item"
+    assert to_do_list_table_cell.text == new_card_name
+    
+def test_can_create_new_to_do_item_and_start_it(driver, test_app):
+    new_card_name = "New Selenium Test Item"
+    driver.get('http://localhost:5000/')
+    new_to_do_textbox = driver.find_element_by_name("todoTitle")
+    new_to_do_textbox.send_keys(new_card_name)
+    save_button = driver.find_element_by_id("save_new")
+    ActionChains(driver).click(save_button).perform()
```

Comment:
ActionChains is for chaining multiple actions in a row. If you're just doing one thing, it's simpler to write `save_button.click()`

And actually this seems to cause me viewport errors while save_button.click doesn't